ibis/bigquery/tests/test_client.py

```diff
@@ -397,3 +397,69 @@ def test_parted_column(client, kind, option, expected_fn):
         t = client.table(table_name)
     expected_column = expected_fn(kind)
     assert t.columns == [expected_column, 'string_col', 'int_col']
+
+
+def test_cross_project_query():
+    con = ibis.bigquery.connect(
+        project_id='ibis-gbq',
+        dataset_id='bigquery-public-data.stackoverflow')
+    table = con.table('posts_questions')
```

Comment:
Could we add another test cross-project query by setting the `database` parameter here and in other methods (like `exists_database`, `exists_table`, `list_tables`).

Reply:
Absolutely, willl do.

Reply:
@tswast How do you feel about disabling cross project versions of these methods? You'd only be able to pass in a database name/table name, list tables/databases in the current `data_project_id`.

Reply:
With the expectation that we'll follow-up to open them up later? Or are you still thinking we'd want multiple backends for that use?

Reply:
I'm thinking that we'd want mulitple clients for that use (multiple backends is a different thing, and somewhat of a can of worms).

Reply:
I'd caution against multiple clients if only because the auth flow can take some time, especially if it's pinging the metadata server for GCE-based credentials.

Reply:
See: https://github.com/pydata/pandas-gbq/pull/127#discussion_r178683518

```
In [4]: %timeit google.auth.default()
573 ms ± 14.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

The Client constructor calls `google.auth.default()` if no credentials are passed in.

Reply:
It can take even longer on some systems (if it is trying to get credentials from GCE metadata server IIRC)

Reply:
Interesting. What is the recommended way to query across projects that doesn't involve creating a new `bq.Client` object.

Reply:
Ah I didn't realize that some of the key methods like dataset take a project parameter. Let me tinker a bit